tests/jobs/scala/src/main/scala/KafkaCassandra.scala

```diff
@@ -0,0 +1,257 @@
+      rdd.foreachPartition { partition =>
+
+        val producer = new KafkaProducer[String, String](kafkaProperties)
+        partition.foreach { word =>
+          logInfo(s"Writing $word to Kafka")
+          val msg = new ProducerRecord[String, String](topic, null, word)
+          producer.send(msg)
+        }
+        producer.close()
+      }
+    }
+
+    streamingContext.start()
+    streamingContext.awaitTermination()
+  }
+
+
+  def getKafkaProperties(brokers : String, kerberized: Boolean) : Properties = {
+
+    val props = new Properties()
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokers)
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer")
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringSerializer")
+    if (kerberized) {
+      props.put("sasl.kerberos.service.name", "kafka")
+      props.put("security.protocol", "SASL_PLAINTEXT")
+      props.put("sasl.mechanism", "GSSAPI")
+    }
+
+    props
+  }
+
+}
+
+
+object KafkaWordCount extends Logging {
+  def main(args: Array[String]) : Unit = {
+    StreamingExamples.setStreamingLogLevels()
+
+    val config : Config = Config()
+    if (StreamingExamples.getParser( this.getClass.toString(), false ).parse(args, config).isEmpty) {
+      logError("Bad arguments")
+      System.exit(1)
+    }
+
+    val spark = StreamingExamples.createSparkSession(config)
+
+    val streamingContext = new StreamingContext(spark.sparkContext, Seconds(config.batchSizeSeconds))
+
+    // Create direct kafka stream with brokers and topics
+    val topicsSet = config.topics.split(",").toSet
+
+    val kafkaParams = getKafkaProperties(config.brokers, config.groupId, config.kerberized)
+
+    val cassandraEnabled : Boolean = config.cassandra
+
+    // Set up the cassandra session:
+    val keyspaceName = config.cassandraKeyspace
+    val tableName = config.cassandraTable
+    val cassandraColumns = SomeColumns("word" as "_1", "ts" as "_2", "count" as "_3")
+
+    if (cassandraEnabled) {
+      CassandraConnector(spark.sparkContext.getConf).withSessionDo { session =>
+
+        session.execute(
+          s"""DROP KEYSPACE IF EXISTS $keyspaceName""")
+        session.execute(
+          s"""CREATE KEYSPACE $keyspaceName WITH REPLICATION = {
+            'class': 'SimpleStrategy',
+            'replication_factor': 3
+          }""")
+        session.execute(
+          s"""
+            CREATE TABLE IF NOT EXISTS $keyspaceName.$tableName (
+              ts timestamp,
+              word text,
+              count int,
+              PRIMARY KEY(word, ts)
+            );"""
+        )
+      }
+    }
+
+    val messages = KafkaUtils.createDirectStream[String, String](
+      streamingContext,
+      LocationStrategies.PreferConsistent,
+      ConsumerStrategies.Subscribe[String, String](topicsSet, kafkaParams))
+
+    val timestamp = new Date()getTime()
```

Comment:
`(new Date()).getTime()`?

Also `getTime()` returns a long, do you know if the cassandra driver deals with longs for timestamp columns?